File: php/testcode/benchmark_test_00615.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00615(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->param('data');
    $hash = sodium_bin2hex(sodium_crypto_generichash($data));
    return BenchmarkResponse::ok($hash);
}

File: rust/testcode/benchmark_test_00615.rs
fn mongo_find(collection: &str, filter: &str) -> String {
    format!("Query on {} with filter: {}", collection, filter)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut user_filter = format!("{{\"name\": \"{}\"}}", req.param("name"));
    user_filter = "{\"verified\":true}".to_string();
    let result = mongo_find("accounts", &user_filter);
    super::shared::BenchmarkResponse::ok(&result)
}
